dll/win32/shell32/CDefaultContextMenu.cpp

```diff
@@ -1141,6 +1158,52 @@ CDefaultContextMenu::InvokeRegVerb(
 
     PStaticShellEntry pEntry = &m_StaticEntries.GetAt(it);
 
+    HKEY hVerbKey;
```

Comment:
I would prefer a `CRegKey` here, because the control flow is already starting to grow quite a bit.

Reply:
Then you can also ditch the `Done`.

Reply:
> Then you can also ditch the `Done`

No I cannot, that logic needs to stay. I will add a comment. If CoCreateInstance is successful, it must return even if the action itself fails, Windows never falls back to the classic command invocation.